cms/djangoapps/contentstore/views/entrance_exam.py

```diff
@@ -0,0 +1,206 @@
+    }
+    CourseMetadata.update_from_dict(metadata, course, request.user)
+
+    # Add an entrance exam milestone if one does not already exist
+    namespace = '{}.entrance_exams'.format(unicode(course_key))
+    milestones = milestones_api.get_milestones(namespace)
+    if len(milestones):
+        milestone = milestones[0]
+    else:
+        milestone = milestones_api.add_milestone({
+            'name': 'Completed Entrance Exam'.format(uuid4()),
+            'namespace': '{}.entrance_exams'.format(unicode(course_key)),
+            'description': 'Autogenerated during {} entrance exam creation workflow.'.format(
+                unicode(course.id)
+            ),
+        })
+    milestones_api.add_course_milestone(
+        unicode(course.id),
+        'requires',
+        milestone
+    )
+    milestones_api.add_course_content_milestone(
+        unicode(course.id),
+        created_item['locator'],
+        'fulfills',
+        milestone
+    )
+
+    return HttpResponse(status=201)
+
+
+def _get_entrance_exam(request, course_key):  # pylint: disable=W0613
+    """
+    Internal workflow operation to retrieve an entrance exam
+    """
+    course = modulestore().get_course(course_key)
+    if course is None:
+        return HttpResponse(status=400)
+    if not getattr(course, 'entrance_exam_id'):
+        return HttpResponse(status=404)
+    try:
+        exam_key = UsageKey.from_string(course.entrance_exam_id)
+    except InvalidKeyError:
+        return HttpResponse(status=404)
+    try:
+        exam_descriptor = modulestore().get_item(exam_key)
+        return HttpResponse(
+            _serialize_entrance_exam(exam_descriptor),
+            status=200, mimetype='application/json')
+    except ItemNotFoundError:
+        return HttpResponse(status=404)
+
+
+def delete_entrance_exam(request, course_key):
+    """
+    api method to delete an entrance exam
+    """
+    return _delete_entrance_exam(request=request, course_key=course_key)
+
+
+def _delete_entrance_exam(request, course_key):
+    """
+    Internal workflow operation to remove an entrance exam
+    """
+    course = modulestore().get_course(course_key)
+    if course is None:
+        return HttpResponse(status=400)
+
+    children = modulestore().get_items(course_key, qualifiers={'metadata.is_entrance_exam': True})
+    if len(children):
+        for child in children:
+            delete_item(request, child.location)
+            milestones_api.remove_content_references(unicode(child.location))
+
+    # Reset the entrance exam flags on the course
+    # Reload the course so we have the latest state
+    course = modulestore().get_course(course_key)
+    if getattr(course, 'entrance_exam_id'):
+        metadata = {
+            'entrance_exam_enabled': False,
+            'entrance_exam_minimum_score_pct': None,
+            'entrance_exam_id': None,
+        }
+        CourseMetadata.update_from_dict(metadata, course, request.user)
```

Comment:
likewise, I think we can use the normal xBlock field setters directly to make this a bit simplier